db/system_keyspace.cc

```diff
@@ -938,11 +938,11 @@ schema_ptr system_keyspace::discovery() {
         return schema_builder(NAME, DISCOVERY, id)
             // This is a single-partition table with key 'peers'
             .with_column("key", utf8_type, column_kind::partition_key)
-            // Opaque connection properties. See `raft::server_info`.
-            .with_column("server_info", bytes_type, column_kind::clustering_key)
+            // Peer ip address
+            .with_column("ip", inet_addr_type, column_kind::clustering_key)
```

Comment:
When we discussed this, you suggested using a string type so we wouldn't have to change the schema of this table when/if we switch to URIs. `bytes_type` would probably also do the job, which means you shouldn't have to do this particular change at all, there was nothing wrong with `server_info` using `bytes_type`.

So WDYT about:
- leaving the schema as it is, but the comment needs to be updated (because now it refers to `raft::server_info`)
- or changing it to a string type?

Reply:
I already researched this idea - immediately after we discussed it. 
`kostja@hulk:~/work/scylla/scylla$ grep inet_addr_type db/system_keyspace.cc | grep with_column
            .with_column("ip_addr", inet_addr_type)
            .with_column("ip_addr", inet_addr_type, column_kind::clustering_key)
            .with_column("peer", inet_addr_type, column_kind::partition_key)
            .with_column("endpoint", inet_addr_type, column_kind::clustering_key)
            .with_column("address", inet_addr_type, column_kind::partition_key)
`
We use inet_addr_type in many, critical places. peers, raft config (the most critical one), local, peer_events, cluster_status, token_ring, clients, we use it for rpc addresses, broadcast addresses, peer addresses, listen addresses. In these circumstances switching discovery to a different scheme is just adding diversity which will never be appreciated. At the very least it should be along with raft_config and as a separate patch.